Makefile

```diff
@@ -8,10 +8,10 @@ build:
 	cd ${VDIR} && go build ./...
 
 test:
-	cd ${VDIR} && go test ./...
+	cd ${VDIR} && go test -tags=all ./...
```

Comment:
FYI... The go core unit test bucket has grown a bit over time and some of the tests (e.g. authenticator tests, "context"-related tests, etc.) are "slow" (relatively speaking).  Up to this point, I've typically just used the `-run=<regexp>` option on the `go test` command to run only a subset of the tests while doing development.   But, as part of this PR, I've introduced the use of ginkgo/gomega for the new retry-related tests since that seemed the most natural way to write them (similar to our generated unit tests, etc.), and that means that the `-run=<regexp>` option is no longer as useful as it was.   So, I went ahead and added build flags to each of the unit test files to sort of classify the sets of tests.  All _test.go files have the tag "all" so that when we run tests with `-tags=all` they all get run.    Other tags that were added are "fast", "slow", "auth", "basesvc", "log" and "retries".   To run only the "fast" tests, you'd use `go test -tags=fast`.   To run only the retry-related tests, use `go test -tags=retries`, etc.
The Makefile (used in Travis and in local sandboxes) will continue to run all the tests.